superset-frontend/src/components/Button/index.tsx

```diff
@@ -200,14 +200,16 @@ export default function Button(props: ButtonProps) {
         },
         '&[disabled], &[disabled]:hover': {
           color: grayscale.base,
-          backgroundColor: backgroundColorDisabled,
-          borderColor: borderColorDisabled,
+          backgroundColor:
+            buttonStyle === 'link' ? 'transparent' : backgroundColorDisabled,
+          borderColor:
+            buttonStyle === 'link' ? 'transparent' : borderColorDisabled,
         },
         marginLeft: 0,
         '& + .superset-button': {
           marginLeft: theme.gridUnit * 2,
         },
-        '& :first-of-type': {
+        '& > :first-of-type': {
```

Comment:
Can you explain why this was needed? I'm just trying to understand if this change can impact other buttons in the application.

Reply:
This selector was targeting every "first of type" element being a descendant of `Button`. In this case (see screenshot below), it would target both `<span>` and `<svg>` elements, so in result we'd apply margin twice. With this change, we'll only target direct children of `Button`.

<img width="549" alt="image" src="https://user-images.githubusercontent.com/15073128/154531575-e15b41c3-4c69-4c0a-877f-7f3f38a4c527.png">

Reply:
Understood. I just don't know if this change can cause a regression elsewhere.